knimin/lib/data_access.py

```diff
@@ -2399,64 +2400,205 @@ def getProjectNames(self):
+        return True
+
+    def create_plate(self, name, email, plate_type):
+        """Creates a new plate and sets values for mandatory fields
+
+        Parameters
+        ----------
+        name : str
+        email : str
+        plate_type : str
+
+        Returns
+        -------
+        int
+            Id of the created plate
+        """
+        sql = """INSERT INTO pm.plate (name, email, plate_type_id)
+                 VALUES (%s, %s, (SELECT plate_type_id FROM pm.plate_type
+                 WHERE name = %s)) RETURNING plate_id"""
+        sql_args = [name, email, plate_type]
+        return self._con.execute_fetchone(sql, sql_args)[0]
+
+    def delete_plate(self, plate_ids):
+        """Deletes plates and corresponding plate-to-sample maps
+
+        Parameters
+        ----------
+        plate_ids : list of int
+            Plate ids to delete
+
+        Returns
+        -------
+        int
+            Number of plates deleted
+        """
+        sql = """DELETE FROM pm.plate_sample WHERE plate_id = %s"""
+        self._con.executemany(sql, [[x] for x in plate_ids])
+        sql = """DELETE FROM pm.plate WHERE plate_id = %s"""
+        self._con.executemany(sql, [[x] for x in plate_ids])
+        return len(plate_ids)
 
     def set_plate_info(self, plate_id, plate_info):
         """Sets attributes of a plate by id
 
         Parameters
         ----------
         plate_id : int
-            Existing plate id (>0) or create new plate (0)
-        plate_info : dict
-            Plate attributes
+            Id of the plate to set attributes to
+        plate_info: tuple of (str x 12)
+            name, template, linker_seq, extraction_kit_lot, extraction_robot,
+            tm1000_8_tool, master_mix_lot, water_lot_id, processing_robot_id,
+            tm300_8_tool_id, tm50_8_tool_id, notes
 
         Returns
         -------
-        int
-            Plate id of the created / modified plate
+        bool
+            True if successful
+
+        Notes
+        -----
+        Empty string or None values set corresponding columns to NULL
         """
-        keys = ', '.join([str(x) for x in plate_info.keys()])
-        values = ', '.join(['\'' + str(x) + '\'' for x in plate_info.values()])
-        sql = ''
-        if plate_id == 0:  # Create new plate
-            sql = 'INSERT INTO pm.plate (' + keys + ') VALUES (' + values + ')'
-        else:  # Modify existing plate
-            sql = 'UPDATE pm.plate SET (' + keys + ') = (' + values + ')'
-            sql += ' WHERE plate_id = ' + str(plate_id)
-        sql += ' RETURNING plate_id'
-        return self._con.execute_fetchone(sql)[0]
+        sql = """UPDATE pm.plate
+                 SET name = %s,
+                     template_id = (SELECT template_id
+                        FROM pm.template WHERE name = %s),
+                     linker_seq = %s,
+                     extraction_kit_lot_id = (SELECT extraction_kit_lot_id
```

Comment:
I don't know how JOIN works in an UPDATE command. @wasade can you give me an example?